### ISSUE TITLE
Notify of redundant `@Ignore` annotations

### ISSUE THREAD

**Issue by Greedquest** [rubberduck-vba/Rubberduck#5518]
I was doing some tidying and found this comment:

    '@Ignore StopKeyword: For Debugging, once RD understands code paths this Ignore won't be necessary

It got me thinking, RD needs some way of telling me when an Ignore annotation is no longer necessary

**Description:**
A new meta-inspection perhaps, that flags `@Ignore[Module]` statements which no longer have any effect - perhaps because the code has changed or because RD's capabilities have improved and you no longer need to @Igonore that false positive.

**Justification:**
One reason is that it will reduce clutter from pointless `@Ignore`s which have been made redundant as RD's capabilities improve - as well as keeping users in the loop about progress with false positives.

Anothers situation where implementing this might be useful is where a user ignores an inspection (with good reason). They subsequently change their code so that the Ignore is no longer necessary, but now any further changes they make will not trigger the inspection since it's still being ignored by an out of date flag - preventing RD from giving what would now be considered valid results. This is especially a problem with module level ignores.

I would assume a system like this takes account of the parameters to the Ignore annotation and has a quickfix that allows you to remove them when their presence or absence no longer has any impact on the inspection results

**Comment by Greedquest** [comment]
If this is implemented as another inspection result, it would be assumed that it too could be ignored - if this is done on a case by case basis then it needs to annotate an `'@Ignore` comment/argument, not a module or statement.

**Comment by retailcoder** [comment]
I think this would need to be a "Rubberduck Opportunities" meta-inspection - IIRC there's a flag for non-annotatable inspection results (currently only *DefaultProjectName* uses it) that could be handy here.

Basically the inspection would be flagging `@Ignore` and `@IgnoreModule` annotations that aren't muting any inspection results.... and doing that would need to change how RD tracks inspection results; ignored results are simply filtered-out/discarded... and no inspection currently has access to other inspections' results.

This one needs some serious thought.

**Comment by daFreeMan** [comment]
Yeah, I'd like this one, too. I've found quite a number of false positive inspection results over the years and have put in `'@Ignore <inspection>` to clean up, but the error has since been fixed and they're now useless.

Somewhere, I thought I'd submitted an issue to be able to have the `@` annotations show up in the ToDo explorer (when configured by the user), which is what I was intending to use to find these no longer necessary `@Ignores`, but now I can't find that issue. This _would_ be a better fix.
